copilot: Allow enterprise to sign in and use copilot

imumesh18 opened the pull request:
This addresses: https://github.com/zed-industries/zed/pull/32248#issuecomment-2952060834.

This PR address two main things one allowing enterprise users to use copilot chat and completion while also introducing the new way to handle copilot url specific their subscription. Simplifying the UX around the github copilot and removes the burden of users figuring out what url to use for their subscription.

- [x] Pass enterprise_uri to copilot lsp so that it can redirect users to their enterprise server. Ref: https://github.com/github/copilot-language-server-release#configuration-management
- [x] Remove the old ui and config language_models.copilot which allowed users to specify their copilot_chat specific endpoint. We now derive that automatically using token endpoint for copilot so that we can send the requests to specific copilot endpoint for depending upon the url returned by copilot server.
- [x] Tested this for checking the both enterprise and non-enterprise flow work. Thanks to @theherk for the help to debug and test it. 
- [ ] Udpdate the zed.dev/docs to refelect how to setup enterprise copilot.

What this doesn't do at the moment:

* Currently zed doesn't allow to have two seperate accounts as the token used in chat is same as the one generated by lsp. After this changes also this behaviour remains same and users can't have both enterprise and personal copilot installed.

P.S: Might need to do some bit of code cleanup and other things but overall I felt this PR was ready for atleast first pass of review to gather feedback around the implementation and code itself.


Release Notes:

- Add enterprise support for GitHub copilot

imumesh18 committed:
copilot: Add enterprise_uri support to sign in flow

imumesh18 commented:
@theherk would be interested in testing this please. This PR is not ready but show work just fine. You can test this by adding the enterprise url in the settings.json. Let me know if it works for you.

```
  "edit_predictions": {
	....
    "copilot": {
	....
	"enterprise_uri": "https://foo.ghe.com"
	
    }
  },
```

theherk commented:
Also, the predictions don't seem to be working, and I'm getting `2025-06-07T13:04:09+02:00 ERROR [project] from extension "GitHub MCP Server" version 0.0.3: missing `github_personal_access_token` setting`.

imumesh18 commented:
Hey @theherk pinged on discord for some help. yes zed login shouldn't be affected and this changes should make it possible to do 1 and 2. If you can help me with a ghe url which is accessible via the public internet so that I can test this that would be great.

imumesh18 committed:
code cleanup

Signed-off-by: Umesh Yadav <git@umesh.dev>

imumesh18 committed:
Refactor Copilot Chat to use configuration struct

- Replace CopilotChatSettings with CopilotChatConfiguration - Remove manual
API/Auth/Models URL settings in favor of enterprise URI - Update initialization
and settings propagation to use new configuration - Simplify Copilot Chat
configuration UI and settings handling

imumesh18 committed:
📎

imumesh18 committed:
Merge branch 'main' into push-vrnmlpplvkql

theherk reviewed (approved):
I have verified that the latest changes works as expected with Github Enterprise Copilot for both edit predictions and chat.

imumesh18 committed:
Refactor Copilot configuration handling for clarity and reuse

- Build configuration objects directly with `json!` macro - Simplify and
deduplicate proxy and enterprise URI settings - Ensure Copilot Chat receives
updated configuration - Send configuration update after LSP server startup

imumesh18 committed:
Merge branch 'main' into push-vrnmlpplvkql

NathanBrodin commented:
Hey, can we get some documentation in [Zed](https://zed.dev/docs/ai/configuration) on how to set up Github Enterprise Copilot?

This PR says it has been implemented, but there is nothing to document how to do it, so I am stuck.

Thanks!
